src/components/AdditionalSpecies.jsx

```diff
@@ -0,0 +1,196 @@
+
+    setSpecies(newTotalData);
+  };
+
+  const handleAllChecked = () => {
+    const newCheckedData = [...species];
+    if (allChecked) {
+      for (let i = 1; i < newCheckedData.length; i += 1) {
+        newCheckedData[i].isChecked = false;
+      }
+    } else {
+      for (let i = 1; i < newCheckedData.length; i += 1) {
+        newCheckedData[i].isChecked = true;
+      }
+    }
+    setDisabled(!newCheckedData.some(findChecked));
+    setAllChecked(!allChecked);
+    setSpecies(newCheckedData);
+  };
+
+  const handleRowCheckedItems = id => {
+    const newCheckedData = [...species];
+    newCheckedData[id].isChecked = !newCheckedData[id].isChecked;
+    setDisabled(!newCheckedData.some(findChecked));
+    setSpecies(newCheckedData);
+  };
+
+  const handleAddRow = () => {
+    if (option !== '') {
+      const newData = {
+        name: option.label,
+        total: 0,
+        isChecked: false,
+        isVisible: 'visible',
+        isDisabled: false,
+      };
+      const newSpecies = [...species, newData];
+
+      setOption('');
+      setSpecies(newSpecies);
+    }
+  };
+
+  const handleDeleteRows = () => {
+    const newSpecies = species.filter(row => !row.isChecked);
+    setDisabled(!newSpecies.some(findChecked));
+    setSpecies(newSpecies);
+  };
+
+  const createTable = m => {
+    return m.map((row, index) => (
+      <Tr height="72px" key={nanoid()}>
+        <Td paddingRight="8px">
+          <Checkbox
+            size="md"
+            isChecked={row.isChecked}
+            onChange={() => handleRowCheckedItems(index)}
+            visibility={row.isVisible}
+            isDisabled={row.isDisabled}
+          />
+        </Td>
+        <Td>
+          <Input size="lg" backgroundColor="#EDF2F7" value={row.name} isReadOnly />
+        </Td>
+        <Td>
+          <NumberInput size="lg" value={row.total} onChange={v => handleAddTotalChange(v, index)}>
+            <NumberInputField />
+          </NumberInput>
+        </Td>
+      </Tr>
+    ));
+  };
+
+  return (
+    <Container maxW="container.xl">
+      <VStack w="full" h="full" spacing="29px" alignItems="flex.start">
+        <Text fontWeight="600" fontSize="2xl">
+          Additional Species
+        </Text>
+        <FormControl>
+          <FormLabel fontWeight="600">Search for a Species:</FormLabel>
+          <SimpleGrid columns={2} h="166px" columnGap="26px">
+            <GridItem colSpan={1}>
+              <SimpleGrid columns={6} h="100%" columnGap="9px">
+                <GridItem colSpan={5}>
+                  <VStack w="full" h="100%" position="relative" alignItems>
+                    <Select value={option} options={options} onChange={v => setOption(v)} />
+                    <Button
+                      w="full"
+                      position="absolute"
+                      bottom={0}
+                      fontWeight="700"
+                      isDisabled={disabled}
+                      onClick={() => handleDeleteRows()}
```

Comment:
What made you decide to use an arrow function instead of calling the handler function directly?

Reply:
Using an anonymous function like this isn't preferred as it can cause unnecessary rerenders, which hurt performance. Sometimes this method is recommended online because it prevents the function from being called as soon as the component renders, but this can be accomplished by removing the parentheses after the function name ie:
`onClick={handleDeleteRows}`

Read more about this [in the beta React docs](https://beta.reactjs.org/learn/responding-to-events#:~:text=Functions%20passed%20to%20event%20handlers%20must%20be%20passed%2C%20not%20called.%20For%20example%3A).